src/logic/calculate.js

```diff
@@ -0,0 +1,113 @@
+import operate from './operate';
+
+function calculate(data, name) {
+  const {
+    total, operation, next, isCalculated,
+  } = data;
+  switch (name) {
+    case ('+/-'):
+      if (next) {
+        return {
+          total,
+          operation,
+          next: next * -1,
+          isCalculated,
+        };
+      }
+      return {
+        total: total * -1,
+        operation,
+        next,
+        isCalculated,
+      };
+
+    case ('AC'):
+      return {
+        total: null,
+        operation: null,
+        next: null,
+        isCalculated: false,
+      };
+
+    case ('+'):
+    case ('-'):
+    case ('÷'):
+    case ('X'):
+    case ('%'):
+      if (!total) {
+        return {
+          total: name === 'X' ? '1' : '0',
+          operation,
+          next,
+          isCalculated,
+        };
+      } if ((total && operation && !next) || (total && !operation)) {
+        return {
+          total,
+          operation: name,
+          next,
+          isCalculated,
+        };
+      }
+      return {
+        total: operate(total, operation, next),
+        operation: name,
+        next: null,
+        isCalculated: true,
+      };
+
+
+    case ('='):
+      if (next) {
+        return {
+          total: operate(total, operation, next),
+          operation: null,
+          next: null,
+          isCalculated: true,
+        };
+      } if (total && !next) {
+        return {
+          total, operation: null, next: null, isCalculated: true,
+        };
+      }
+      return {
+        total: '0', operation: null, next: null, isCalculated: true,
+      };
+
+
+    default:
+      if (!operation && !isCalculated) {
+        return {
+          total: total === null ? name : total + name,
+          operation,
+          next,
```

Comment:
Let's handle cases when the button clicked is `.` differently. 
Multiple `.` should not be allowed. That is, when the value contains a dot, adding another dot should be ignored and not allowed.

See below:

![dot-bug](https://user-images.githubusercontent.com/36057474/77246212-3a8b0300-6c25-11ea-87a7-dde1426c3bd1.gif)